### PR TITLE
Remove insightIds from critical dashboard loading path

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/42263

## Background
This PR simply removes `insightIds` field from the dashboard GQL query; this information is no longer needed for the dashboard page's initial dashboard request (it was needed before because of the SettingCascade backend data source). It also removes one of the GQL backend fetch method and places the query right next to the UI component that uses this query.

**The outcome of this change**
All Insight dashboard response size was reduced from 12kb to 1.9kb

## Test plan
- Make sure that the dashboard page loads all insights (as it loads on the k8s)
- Check that the initial dashboard query doesn't load `insightIds` field
- Check that add/remove insight modal adds and/or removes insight from the dashboard properly 

## App preview:

- [Web](https://sg-web-vk-improve-dashboard-insight.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ikqwzkbvsm.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx [hunk 1]
```diff
@@ -16,23 +23,64 @@ import {
...
+        }
+
+        accessibleInsights: insightViews {
+            nodes {
+                ...AccessibleInsight
+            }
+        }
+    }
+`
+
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx:58]: Question non-blocking and not needed for this PR.

As we bring the data closer to use like this. Maybe we structure our components like:

/AddInsightModal
  /index.ts <- export all the things
  /AddInsightModal.tsx
  /query.ts <- maybe a more specific name
  /styles.scss

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx [hunk 2]
```diff
@@ -51,36 +99,46 @@ export const AddInsightModal: React.FunctionComponent<React.PropsWithChildren<Ad
...
-
-            {insights.length > 0 && (
-                <AddInsightModalContent
-                    initialValues={initialValues}
-                    insights={insights}
-                    dashboardID={dashboard.id}
-                    onCancel={onClose}
-                    onSubmit={handleSubmit}
-                />
+            {loading && !data && <LoadingSpinner inline={false} />}
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx:109]: Bless you, useQuery. The hero we need, not the hero we deserve.

**Review by unclejustin — APPROVED** [review]
LGTM 🌮

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/add-insight-modal/AddInsightModal.tsx:58]: Agree! I think it's time to gather our approaches in the code insights codebase and capture them somewhere (I also saw that Admin UI does the same structure in their components)

**Pull request merged**